sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Parser.cs

```diff
@@ -0,0 +1,186 @@
+                    IsStatic = method.Modifiers.Any(SyntaxKind.StaticKeyword),
+                    ParentFunctionClass = classInfo
+                };
+
+                functionList.Add(funcInfo);
+            }
+
+            return functionList;
+        }
+
+        private bool IsTaskType(ITypeSymbol typeSymbol)
+        {
+            return
+                SymbolEqualityComparer.Default.Equals(typeSymbol.OriginalDefinition, _knownTypes.TaskType) ||
+                SymbolEqualityComparer.Default.Equals(typeSymbol.OriginalDefinition, _knownTypes.TaskOfTType) ||
+                SymbolEqualityComparer.Default.Equals(typeSymbol.OriginalDefinition, _knownTypes.ValueTaskType) ||
+                SymbolEqualityComparer.Default.Equals(typeSymbol.OriginalDefinition, _knownTypes.ValueTaskOfTTypeOpt);
+        }
+
+        private bool IsReturnValueAssignable(IMethodSymbol methodSymbol)
+        {
+            if (methodSymbol.ReturnsVoid)
+            {
+                return false;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(methodSymbol.ReturnType.OriginalDefinition, _knownTypes.TaskType))
+            {
+                return false;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(methodSymbol.ReturnType.OriginalDefinition,
+                    _knownTypes.ValueTaskType))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static IEnumerable<string> GetConstructorParamTypeNames(ClassDeclarationSyntax functionClass,
+            SemanticModel model)
+        {
+            var firstConstructorMember = GetBestConstructor(functionClass);
+
+            if (firstConstructorMember is not ConstructorDeclarationSyntax constructorSyntax)
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var constructorParamTypeNames = new List<string>(constructorSyntax.ParameterList.Parameters.Count);
+
+            foreach (var param in constructorSyntax.ParameterList.Parameters)
+            {
+                if (model.GetDeclaredSymbol(param) is not IParameterSymbol parameterSymbol)
+                {
+                    continue;
+                }
+
+                // We are getting fully qualified name of the type
+                constructorParamTypeNames.Add(parameterSymbol.Type.ToDisplayString());
+            }
+
+            return constructorParamTypeNames;
+        }
+
+        private static MemberDeclarationSyntax GetBestConstructor(ClassDeclarationSyntax functionClass)
+        {
+            // TO DO: Fix this.
+            // Currently picking first constructor.
+            var firstConstructorMember =
+                functionClass.Members.FirstOrDefault(member => member is ConstructorDeclarationSyntax);
+
+            return firstConstructorMember;
+        }
+    }
+
+    internal class ExecutableFunction
```

Comment:
I have a slight preference towards having this class and `FunctionClass` below it in their own files, but no hard opposition.